### PR TITLE
Add Sha keyboard

### PR DESCRIPTION
**Pull request by skarrmann**
## Description

This PR adds a keyboard definition and default keymap for my new keyboard [Sha](https://github.com/skarrmann/sha).

Note the default keymap uses the RGB Light - Lighting Layers feature to demonstrate how to control the RP2040-Matrix's LEDs. I included this feature since it may be useful for users to have a practical reference implementation for indicating keyboard status with this microcontroller board's LEDs.

Thank you!

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

N/A

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/sha/rules.mk [hunk 1]
```diff
@@ -0,0 +1,2 @@
+RGBLIGHT_ENABLE = yes
+WS2812_DRIVER = vendor
```
Comment (zvecr) [inline keyboards/sha/rules.mk:2]:
```suggestion
# This file intentionally left blank
```

File: keyboards/sha/info.json [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+        "sleep": true
+    },
+    "url": "https://github.com/skarrmann/sha",
+    "usb": {
+        "device_version": "1.0.0",
+        "pid": "0x0000",
+        "vid": "0xFEED"
+    },
+    "ws2812": {
+        "pin": "GP16"
```
Comment (zvecr) [inline keyboards/sha/info.json:39]:
```suggestion
        "driver": "vendor",
        "pin": "GP16"
```

File: keyboards/sha/info.json [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+    "maintainer": "skarrmann",
+    "bootloader": "rp2040",
+    "diode_direction": "COL2ROW",
+    "features": {
+        "bootmagic": true,
+        "command": false,
+        "console": false,
+        "extrakey": true,
+        "mousekey": true,
+        "nkro": true
```
Comment (zvecr) [inline keyboards/sha/info.json:13]:
```suggestion
        "nkro": true,
        "rgblight": true
```

File: keyboards/sha/info.json [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+    },
+    "matrix_pins": {
+        "cols": ["GP8", "GP9", "GP10", "GP11", "GP12", "GP13", "GP14"],
+        "rows": ["GP28", "GP4", "GP27", "GP5", "GP26", "GP6", "GP15", "GP7"]
+    },
+    "processor": "RP2040",
+    "rgblight": {
+        "animations": {},
```
Comment (zvecr) [inline keyboards/sha/info.json:21]:
```suggestion
        "animations": {
            "breathing": true,
            "rainbow_mood": true,
            "rainbow_swirl": true,
            "snake": true,
            "knight": true,
            "christmas": true,
            "static_gradient": true,
            "rgb_test": true,
            "alternating": true,
            "twinkle": true
        },
```

**Review by zvecr — CHANGES_REQUESTED** [review]

**Comment by skarrmann** [comment]
I confirmed moving the noted settings into info.json works on hardware. Thank you @zvecr!

File: keyboards/sha/info.json [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+            "rgb_test": true,
+            "alternating": true,
+            "twinkle": true
+        },
+        "default": {
+            "val": 0
+        },
+        "layers": {
+            "enabled": true
+        },
```
Comment (drashna) [inline keyboards/sha/info.json:39]: This would need to be set at the keymap level, or there needs to be code at the kb level.

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/sha/info.json (hunk 5 above)
Comment (skarrmann) [reply keyboards/sha/info.json]: Thanks @drashna! Enabling RGB Lighting Layers has been removed from the keyboard info.json, and moved into the default keymap config.h. RGB Lighting Layers is now defined solely at the keymap level.

**Review by fauxpark — APPROVED** [review]

**Review by waffle87 — APPROVED** [review]

**Pull request merged**